website/docs/r/cognito_user_pool_client.markdown

```diff
@@ -0,0 +1,57 @@
+layout: "aws"
+page_title: "AWS: aws_cognito_user_pool_client"
+side_bar_current: "docs-aws-resource-cognito-user-pool-client"
+description: |-
+  Provides a Cognito User Pool Client resource.
+
+# aws_cognito_user_pool_client
+
+Provides a Cognito User Pool Client resource.
+
+## Example Usage
+
+### Create a basic user pool client client
+
+```hcl
+resource "aws_cognito_user_pool" "pool" {
+  name = "pool"
+}
+
+resource "aws_cognito_user_pool_client" "client" {
+  name = "client"
+  
+  user_pool_id = "${aws_cognito_user_pool.pool.id}"
+}
+```
+
+### Create a user pool client with no SRP authentication
+```hcl
+resource "aws_cognito_user_pool" "pool" {
+  name = "pool"
+}
+
+resource "aws_cognito_user_pool_client" "client" {
+  name = "client"
+
+  user_pool_id = "${aws_cognito_user_pool.pool.id}"
+
+  generate_secret = true
+  explicit_auth_flows = ["ADMIN_NO_SRP_AUTH"]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the application client.
+* `generate_secret` - (Optional) Should an application secret be generated. AWS JavaScript SDK requires this to be false.
+* `user_pool_id` - (Required) The user pool the client belongs to.
+* `explicit_auth_flows` - (Optional) List of authentication flows (ADMIN_NO_SRP_AUTH, CUSTOM_AUTH_FLOW_ONLY)
+
+## Attribute Reference
+
+The following attributes are exported:
+
+* `id` - The id of the user pool client.
+* `client_secret` - The client secret of the user pool client.
```

Comment:
It looks like this field is missing in the schema and CRUD - do you mind adding it?

Reply:
The client secret is generated and returned by AWS so not sure it needs to be in the CRUD?

Reply:
It doesn't need to be in `CUD`, but in Read for sure - otherwise it will remain empty. More importantly it needs to be present in the schema as `Computed: true` field.